pinot-query-planner/src/main/java/org/apache/pinot/query/planner/explain/PlanNodeSorter.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    @Override
+    public PlanNode visitSetOp(SetOpNode node, Comparator<PlanNode> comparator) {
+      return defaultNode(node, comparator);
+    }
+
+    @Override
+    public PlanNode visitExchange(ExchangeNode node, Comparator<PlanNode> comparator) {
+      return defaultNode(node, comparator);
+    }
+
+    @Override
+    public PlanNode visitExplained(ExplainedNode node, Comparator<PlanNode> comparator) {
+      if (!node.getTitle().contains(ExplainNodeSimplifier.COMBINE) || node.getInputs().size() <= 1) {
+        return defaultNode(node, comparator);
+      }
+      List<PlanNode> simplifiedChildren = new ArrayList<>(applyToChildren(node.getInputs(), comparator));
+      simplifiedChildren.sort(comparator);
+      if (simplifiedChildren.equals(node.getInputs())) {
+        return node;
+      }
+      return node.withInputs(simplifiedChildren);
+    }
+
+    private List<PlanNode> applyToChildren(List<PlanNode> children, Comparator<PlanNode> comparator) {
+      List<PlanNode> simplifiedChildren = null;
+      for (int i = 0; i < children.size(); i++) {
+        PlanNode child = children.get(i);
+        PlanNode newChild = child.visit(this, comparator);
+        if (child != newChild) {
+          if (simplifiedChildren == null) {
+            simplifiedChildren = new ArrayList<>(children);
+          }
+          simplifiedChildren.set(i, newChild);
+        }
+      }
+      return simplifiedChildren != null ? simplifiedChildren : children;
+    }
+  }
+
+  public static class DefaultComparator implements Comparator<PlanNode> {
+    public static final DefaultComparator INSTANCE = new DefaultComparator();
+
+    private DefaultComparator() {
+    }
+
+    @Override
+    public int compare(PlanNode o1, PlanNode o2) {
+      if (!(o1 instanceof ExplainedNode) || !(o2 instanceof ExplainedNode)) {
+        return o1.getClass().getSimpleName().compareTo(o2.getClass().getSimpleName());
+      }
+      ExplainedNode node1 = (ExplainedNode) o1;
+      ExplainedNode node2 = (ExplainedNode) o2;
+
+      int cmp = node1.getTitle().compareTo(node2.getTitle());
+      if (cmp != 0) {
+        return cmp;
+      }
+      TreeSet<String> allKeys = new TreeSet<>(Sets.union(
+          node1.getAttributes().keySet(), node2.getAttributes().keySet()));
+      for (String key : allKeys) {
+        Plan.ExplainNode.AttributeValue value1 = node1.getAttributes().get(key);
+        Plan.ExplainNode.AttributeValue value2 = node2.getAttributes().get(key);
+        if (value1 == null) {
+          // there is an attribute in node2 with a _smaller_ key that is not in node1
+          return -1;
+        }
+        if (value2 == null) {
+          // there is an attribute in node1 with a _smaller_ key that is not in node2
+          return 1;
+        }
+        int cmp2 = value1.getValueCase().compareTo(value2.getValueCase());
+        if (cmp2 != 0) {
+          return cmp2;
+        }
+        int cmp3;
+        switch (value1.getValueCase()) {
```

Comment:
Do we not need to handle the `STRINGLIST` case here?

Reply:
Probably because either I missed it or because JSON (the previous type) is difficult to sort. Adding it now that it is just a list of strings

Reply:
Changed in  https://github.com/apache/pinot/pull/13733/commits/272263125a022bf5419ca85c71d766f6f976a83a